packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -153,6 +156,69 @@ export class Toolkit extends CloudAssemblySourceBuilder implements AsyncDisposab
     };
   }
 
+  /**
+   * Bootstrap Action
+   */
+  public async bootstrap(cx: ICloudAssemblySource, userEnvironmentSpecs: string[], options: BootstrapOptions = {}): Promise<void> {
+    const ioHost = withAction(this.ioHost, 'bootstrap');
+    const assembly = await this.assemblyFromSource(cx);
+
+    let environments;
+    if (userEnvironmentSpecs.length) {
+      environments = environmentsFromDescriptors(userEnvironmentSpecs);
+    } else {
+      // select all environments we can find, from stacks being bootstrapped
+      const stackCollection = assembly.selectStacksV2({
```

Comment:
You can probably move this into the if clause

```suggestion
    let environments;
    if (userEnvironmentSpecs.length) {
      environments = environmentsFromDescriptors(userEnvironmentSpecs);
    } else {
      // select all environments we can find, from stacks being bootstrapped    
      const assembly = await this.assemblyFromSource(cx);
      const stackCollection = assembly.selectStacksV2({
```

Reply:
Done.